DevelopmentProposals/componentConnectInterface/componentConnectInterface.md

```diff
@@ -0,0 +1,62 @@
+## Project: Improved interface for connecting Components
+***OpenSim Development Proposal***
+
+### Overview
+Connecting Components currently looks something like "update the Input of my ConsoleReporter and connect it to this Output from my Model".
+Some users may find it more intuitive to (i) begin with the verb *connect*, and (ii) swap the order of the Input and Output.
+This project proposes to change the user interface for connecting Components while retaining all current functionality.
+See [issue #1118](https://github.com/opensim-org/opensim-core/issues/1118).
+
+### Requirements
+- No change to what can be connected or how the underlying connections are formed.
+- No redundancy in the interface (i.e., the new interface replaces the existing interface).
+
+### Architecture
+Changes to `Component`:
+- Create a new method for forming connections between Components. There will be several variants:
+  - `static void connect(const Object& object, Object& object)` and `static void connect(Object& object, const Object& object)`,
```

Comment:
Components can require a connection to Objects, but Objects cannot require a connection to other Objects. So you would want `static void connect(const Object& object, Component& object)`.